### PR TITLE
RangeV3 update to ConanV2

### PR DESCRIPTION
**Pull request by Nomalah**
Specify library name and version:  **range-v3/**

This is an update to the range-v3 recipe to be conanv2 compatable

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — DISMISSED** [review]

File: recipes/range-v3/all/conanfile.py [hunk 1]
```diff
@@ -35,38 +33,38 @@ def _min_cppstd(self):
 
     def configure(self):
```
Comment (SpaceIm) [inline recipes/range-v3/all/conanfile.py:34]: validate instead

File: recipes/range-v3/all/conanfile.py [hunk 2]
```diff
@@ -35,38 +33,38 @@ def _min_cppstd(self):
...
-                                                    self.settings.compiler,
-                                                    self.settings.compiler.version))
+            self.output.warn(
+                f"{self.settings.compiler} {self.settings.compiler.version} support for range-v3 is unknown, assuming it is supported.")
+        elif Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"range-v3 {self.version} requires C++{self._min_cppstd} with {self.settings.compiler}, which is not supported by {self.settings.compiler} {self.settings.compiler.version}")
 
     def package_id(self):
         self.info.header_only()
```
Comment (SpaceIm) [inline recipes/range-v3/all/conanfile.py:47]:
```suggestion
        self.info.clear()
```

File: recipes/range-v3/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [inline recipes/range-v3/all/conanfile.py:34]:
```suggestion
    def validate(self):
```

File: recipes/range-v3/all/conanfile.py [hunk 3]
```diff
@@ -35,38 +33,38 @@ def _min_cppstd(self):
```
Comment (SpaceIm) [inline recipes/range-v3/all/conanfile.py:33]: add a basic_layout please

File: recipes/range-v3/all/conanfile.py [hunk 4]
```diff
@@ -1,6 +1,9 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.scm import Version
+from conan.tools.files import get, copy
+from conan.tools.build import check_min_cppstd
```
Comment (SpaceIm) [inline recipes/range-v3/all/conanfile.py:5]: add `required_conan_version = ">=1.50.0"`

**Review by SpaceIm — APPROVED** [review]

**Comment by prince-chrismc** [comment]
towards #12888 Was going to use this https://jfrog.zoom.us/webinar/register/WN__ICdV8sATsSJ5oIcBh1XJQ if you want to help with more of them ;)

File: recipes/range-v3/all/conanfile.py [hunk 5]
```diff
@@ -32,41 +34,45 @@ def _min_cppstd(self):
...
-            raise ConanInvalidConfiguration("range-v3 {0} requires C++{1} with {2}, which is not supported by {2} {3}"
-                                            .format(self.version,
-                                                    self._min_cppstd,
-                                                    self.settings.compiler,
-                                                    self.settings.compiler.version))
+            self.output.warn(
+                f"{self.settings.compiler} {self.settings.compiler.version} support for range-v3 is unknown, assuming it is supported.")
+        elif Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"range-v3 {self.version} requires C++{self._min_cppstd} with {self.settings.compiler}, which is not supported by {self.settings.compiler} {self.settings.compiler.version}")
```
Comment (prince-chrismc) [inline recipes/range-v3/all/conanfile.py:54]:
```suggestion
                f"{self.ref} requires C++{self._min_cppstd} with {self.settings.compiler}, which is not supported by {self.settings.compiler} {self.settings.compiler.version}")
```

**Review by prince-chrismc — APPROVED** [review]
One nit but not worth it

**Pull request merged**